specification/sections/02-providers.md

```diff
@@ -150,3 +151,85 @@ class MyProvider implements Provider {
 #### Requirement 2.3.3
 
 > In cases of abnormal execution, the `resolution details` structure's `error message` field **MAY** contain a string containing additional detail about the nature of the error.
+
+### 2.4 Initialization
+
+[![experimental](https://img.shields.io/static/v1?label=Status&message=experimental&color=orange)](https://github.com/open-feature/spec/tree/main/specification#experimental)
+
+#### Requirement 2.4.1
+
+> The `provider` interface **MUST** define an `initialize` function which accepts the global `evaluation context` as a argument, and which can be optionally implemented to perform initialization logic relevant to the provider.
```

Comment:
Is the evaluation context required, or is it optional. Seems that we shouldn't REQUIRE that something be passed here, given that we're now adding more plumbing for updating the context and informing the provider that that's occured.

Could we make it optional and say that if it's not provided then the intial evaulation context defaults to a special-case "empty context", which we define as having no targeting key and no additional fields?

Reply:
I think the problem here is that the sentence is ambiguous.  When I said `and which can be optionally implemented` I was saying the `initialize` function can be optionally implemented by provider authors.

Reply:
I've tried to removed the ambiguity. Please check again this makes sense.